PWGDQ/Tasks/tableReader.cxx

```diff
@@ -766,68 +1026,94 @@ struct AnalysisSameEventPairing {
               fHistMan->FillHistClass(histNames[icut][2].Data(), VarManager::fgValues);
             }
           }
+          // additional histograms with prefiltering
+          if (fPrefilter) {
+            // By default take the bits from table maker, but uses the one newly calculated if we have the process function WithDalitzBits
+            for (int dalitzCut = 0; dalitzCut < nDalitzCuts; dalitzCut++) {
+              if (!((uint8_t(1) << dalitzCut) & twoTracksDalitzFilter)) { // check that none of the track is tagged as dalitz
+                // TrackCut_DalitzCuts histograms
+                if (t1.sign() * t2.sign() < 0) {
+                  fHistMan->FillHistClass(histNames[nTrackCuts * (1 + dalitzCut) + icut][0].Data(), VarManager::fgValues);
+                } else {
+                  if (t1.sign() > 0) {
+                    fHistMan->FillHistClass(histNames[nTrackCuts * (1 + dalitzCut) + icut][1].Data(), VarManager::fgValues);
+                  } else {
+                    fHistMan->FillHistClass(histNames[nTrackCuts * (1 + dalitzCut) + icut][2].Data(), VarManager::fgValues);
+                  }
+                }
+              } // end if not dalitz-rejected
+            }   // end loop dalitz cuts
+          }     // end if prefilter
         } // end if (filter bits)
       }   // end for (cuts)
     }     // end loop over pairs
   }
 
+  void processJpsiToEESkimmedDalitzBits(soa::Filtered<MyEventsVtxCovSelected>::iterator const& event, soa::Filtered<MyBarrelTracksSelected> const& tracks, DalitzBitsReduced const& dalitzBits)
+  {
+    // Reset the fValues array
+    VarManager::ResetValues(0, VarManager::kNVars);
+    VarManager::FillEvent<gkEventFillMap>(event, VarManager::fgValues);
+    runSameEventPairing<VarManager::kJpsiToEE, gkEventFillMap, gkTrackFillMapDalitzBits>(event, tracks, tracks, dalitzBits);
+  }
   void processJpsiToEESkimmed(soa::Filtered<MyEventsVtxCovSelected>::iterator const& event, soa::Filtered<MyBarrelTracksSelected> const& tracks)
   {
     // Reset the fValues array
     VarManager::ResetValues(0, VarManager::kNVars);
     VarManager::FillEvent<gkEventFillMap>(event, VarManager::fgValues);
-    runSameEventPairing<VarManager::kJpsiToEE, gkEventFillMap, gkTrackFillMap>(event, tracks, tracks);
+    runSameEventPairing<VarManager::kJpsiToEE, gkEventFillMap, gkTrackFillMap>(event, tracks, tracks, nullptr);
```

Comment:
as per the previous comments, remove the extra argument. same for all process functions